### DISCUSSION TITLE
making custom filter rule protocol indepent

### DISCUSSION THREAD

**Discussion by eddydc** [AdguardTeam/AdGuardHome#3265] [Q&A]
Hi,
I wanted to block a domain and followed the example 
||example.org^ – block access to the example.org domain and all its subdomains

After implementation, https://example.org isn't blocked. What should I do that everything for that domain gets blocked?
Cheers,
Eddy

**Comment by ameshkov** [comment]
This rule of yours blocks example.org. If it's not blocked in your browser, maybe the problem is with how your devices are configured. Maybe they're not using AGH for instance.

**Reply by eddydc** [reply]
I defined it the admin page of AGH and there is a test option. When one tests for http://example.org it says it is blocked. That's not the case for https://example.org and thus one needs to define two rules for the same domain.

**Reply by ameshkov** [reply]
The test option accepts domain names, not URLs. The fact that it accepted `http://example.org` is a mistake.

**Reply by eddydc** [reply]
OK I understand that the test should only test domains independent from the protocol and the interface is incorrect concerning this. But I encountered several times the issue that https:// isn't blocked and shown on the device despite that AGH is the only DNS server specified.

**Reply by eddydc** [reply]
I've been testing on two machines. One connected to adguard only and one to pihole only. Both are set with the custom blocking rules in their respective formats. Whereas pihole does block the desired domains, AGH doesn't. I thought it was protocol dependent,  but that's not the case. So as far I am concerned, it's a bug.

**Reply by ameshkov** [reply]
I am not sure how exactly you're testing, but `||example.org^` blocks `example.org` regardless of the protocol (DNS knows nothing about protocols anyway).

You may want to use `nslookup` to test this.

**Reply by eddydc** [reply]
It seems that the browser is the culprit. After clearing all caches, things work as expected
